Unable to Build Project Using KVS Producer SDK as Dependency

I'm currently trying to integrate this SDK in a separate application but am having a lot of trouble building it, even with `-DBUILD_DEPENDENCIES` on.

My project is built with CMake. I've added this repository as a submodule (recursively) and added it as a subdirectory. All of the third-party dependencies (curl, log4cplus, openssl) build correctly, but it seems as though compilation of the library itself fails due to the inclusion of system libraries that do not exist. Below you'll find the relevant section of my `CMakeLists.txt`, the output from running CMake from my project's build directory, and finally the output of `make`:

## Relevant CMake Section
```cmake
...
add_subdirectory(dependencies/amazon-kinesis-video-streams-producer-sdk-cpp)
...
```

## CMake Output
```
-- Boost version: 1.65.1
-- Found the following Boost libraries:
--   system
--   filesystem
--   exception
-- Found AWS SDK for C++, Version: 1.7.354, Install Root:/usr/local, Platform Prefix:, Platform Dependent Libraries: pthread;crypto;ssl;z;curl
-- Components specified for AWSSDK: s3, application wll be depending on libs: aws-cpp-sdk-s3;aws-cpp-sdk-core
-- Try finding aws-cpp-sdk-core
-- Found aws-cpp-sdk-core
-- Try finding aws-cpp-sdk-s3
-- Found aws-cpp-sdk-s3
CMake Warning (dev) at dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/CMakeLists.txt:8 (project):
  Policy CMP0048 is not set: project() command manages VERSION variables.
  Run "cmake --help-policy CMP0048" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.

  The following variable(s) would be set to empty:

    PROJECT_VERSION
    PROJECT_VERSION_MAJOR
    PROJECT_VERSION_MINOR
    PROJECT_VERSION_PATCH
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Kinesis Video Cpp Producer path is /path/to/my/repo
-- dependencies install path is /path/to/my/repo/dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/open-source
-- Begin building dependencies.
-- autoconf already built
-- automake already built
-- log4cplus already built
-- Finished building dependencies.
-- Configuring done
-- Generating done
-- Build files have been written to: /path/to/my/repo/dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/dependency/libkvscproducer
[ 11%] Performing update step for 'libkvscproducer-download'
[ 22%] No configure step for 'libkvscproducer-download'
[ 33%] No build step for 'libkvscproducer-download'
[ 44%] No install step for 'libkvscproducer-download'
[ 55%] No test step for 'libkvscproducer-download'
[ 66%] Completed 'libkvscproducer-download'
[100%] Built target libkvscproducer-download
CMake Warning (dev) at dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/dependency/libkvscproducer/kvscproducer-src/CMakeLists.txt:4 (project):
  Policy CMP0048 is not set: project() command manages VERSION variables.
  Run "cmake --help-policy CMP0048" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.

  The following variable(s) would be set to empty:

    PROJECT_VERSION
    PROJECT_VERSION_MAJOR
    PROJECT_VERSION_MINOR
    PROJECT_VERSION_PATCH
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Kinesis Video Producer path is /path/to/my/repo/dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/dependency/libkvscproducer/kvscproducer-src
-- Begin building dependencies.
-- openssl already built
-- curl already built
-- Finished building dependencies.
-- Configuring done
-- Generating done
-- Build files have been written to: /path/to/my/repo/dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/dependency/libkvscproducer/kvscproducer-src/dependency/libkvspic
[ 11%] Performing update step for 'libkvspic-download'
[ 22%] No configure step for 'libkvspic-download'
[ 33%] No build step for 'libkvspic-download'
[ 44%] No install step for 'libkvspic-download'
[ 55%] No test step for 'libkvspic-download'
[ 66%] Completed 'libkvspic-download'
[100%] Built target libkvspic-download
CMake Warning (dev) at dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/dependency/libkvscproducer/kvscproducer-src/dependency/libkvspic/kvspic-src/CMakeLists.txt:4 (project):
  Policy CMP0048 is not set: project() command manages VERSION variables.
  Run "cmake --help-policy CMP0048" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.

  The following variable(s) would be set to empty:

    PROJECT_VERSION
    PROJECT_VERSION_MAJOR
    PROJECT_VERSION_MINOR
    PROJECT_VERSION_PATCH
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Found OpenSSL: /path/to/my/repo/dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/open-source/local/lib/libcrypto.so (found version "1.1.1g")
-- Bitness 64 bits
-- Found OpenSSL: /path/to/my/repo/dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/open-source/local/lib/libcrypto.so (found suitable version "1.1.1g", minimum required is "1")
found log4cplus.
-- adding symlinks
-- Configuring done
-- Generating done
-- Build files have been written to: /path/to/my/repo/release
```

## make
```
[ 17%] Built target kvspicUtils
[ 25%] Built target kvsCommonCurl
[ 57%] Built target kvspic
[ 67%] Built target cproducer
[ 67%] Building CXX object dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/CMakeFiles/KinesisVideoProducer.dir/src/Auth.cpp.o
In file included from /home/automotus/picard/optix/dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/src/Auth.cpp:1:0:
/home/automotus/picard/optix/dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/src/Auth.h:12:10: fatal error: Logger.h: No such file or directory
 #include <Logger.h>
          ^~~~~~~~~~
compilation terminated.
dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/CMakeFiles/KinesisVideoProducer.dir/build.make:62: recipe for target 'dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/CMakeFiles/KinesisVideoProducer.dir/src/Auth.cpp.o' failed
make[2]: *** [dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/CMakeFiles/KinesisVideoProducer.dir/src/Auth.cpp.o] Error 1
CMakeFiles/Makefile2:90: recipe for target 'dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/CMakeFiles/KinesisVideoProducer.dir/all' failed
make[1]: *** [dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/CMakeFiles/KinesisVideoProducer.dir/all] Error 2
Makefile:129: recipe for target 'all' failed
make: *** [all] Error 2
```

It seems to have something to do with the unusual inclusion style of header files in the SDK. Why are project headers included using angle brackets? I've tried to install the library but it seems as though the install step is not configured.

Am I doing something wrong? Is there a recommended method for using this SDK as a dependency? I had no trouble building the GStreamer plugin, but it doesn't satisfy my use-case. I have to use the producer library directly but building it as a dependency has proven quite difficult.

## Updates
It seems as though it's not only the system-style includes causing problems. Running `make -j8` yields a few more errors with standard includes:

```
[ 87%] Building CXX object dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/CMakeFiles/KinesisVideoProducer.dir/src/Auth.cpp.o
[ 88%] Building CXX object dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/CMakeFiles/KinesisVideoProducer.dir/src/CachingEndpointOnlyCallbackProvider.cpp.o
[ 90%] Building CXX object dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/CMakeFiles/KinesisVideoProducer.dir/src/KinesisVideoProducer.cpp.o
[ 91%] Building CXX object dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/CMakeFiles/KinesisVideoProducer.dir/src/DefaultCallbackProvider.cpp.o
[ 92%] Building CXX object dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/CMakeFiles/KinesisVideoProducer.dir/src/KinesisVideoStream.cpp.o
[ 92%] Building CXX object dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/CMakeFiles/KinesisVideoProducer.dir/src/common/PutFrameHelper.cpp.o
[ 93%] Building CXX object dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/CMakeFiles/KinesisVideoProducer.dir/src/credential-providers/RotatingCredentialProvider.cpp.o
[ 93%] Building CXX object dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/CMakeFiles/KinesisVideoProducer.dir/src/credential-providers/IotCertCredentialProvider.cpp.o
In file included from /path/to/my/repo/dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/src/common/PutFrameHelper.cpp:1:0:
/path/to/my/repo/dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/src/common/PutFrameHelper.h:4:10: fatal error: KinesisVideoProducer.h: No such file or directory
 #include "KinesisVideoProducer.h"
          ^~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/CMakeFiles/KinesisVideoProducer.dir/build.make:302: recipe for target 'dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/CMakeFiles/KinesisVideoProducer.dir/src/common/PutFrameHelper.cpp.o' failed
make[2]: *** [dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/CMakeFiles/KinesisVideoProducer.dir/src/common/PutFrameHelper.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
In file included from /path/to/my/repo/dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/src/credential-providers/RotatingCredentialProvider.cpp:1:0:
/path/to/my/repo/dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/src/credential-providers/RotatingCredentialProvider.h:4:10: fatal error: Auth.h: No such file or directory
 #include "Auth.h"
          ^~~~~~~~
compilation terminated.
dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/CMakeFiles/KinesisVideoProducer.dir/build.make:350: recipe for target 'dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/CMakeFiles/KinesisVideoProducer.dir/src/credential-providers/RotatingCredentialProvider.cpp.o' failed
make[2]: *** [dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/CMakeFiles/KinesisVideoProducer.dir/src/credential-providers/RotatingCredentialProvider.cpp.o] Error 1
In file included from /path/to/my/repo/dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/src/credential-providers/IotCertCredentialProvider.cpp:1:0:
/path/to/my/repo/dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/src/credential-providers/IotCertCredentialProvider.h:5:10: fatal error: Auth.h: No such file or directory
 #include <Auth.h>
          ^~~~~~~~
compilation terminated.
dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/CMakeFiles/KinesisVideoProducer.dir/build.make:326: recipe for target 'dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/CMakeFiles/KinesisVideoProducer.dir/src/credential-providers/IotCertCredentialProvider.cpp.o' failed
make[2]: *** [dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/CMakeFiles/KinesisVideoProducer.dir/src/credential-providers/IotCertCredentialProvider.cpp.o] Error 1
In file included from /path/to/my/repo/dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/src/Auth.cpp:1:0:
/path/to/my/repo/dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/src/Auth.h:12:10: fatal error: Logger.h: No such file or directory
 #include <Logger.h>
          ^~~~~~~~~~
compilation terminated.
dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/CMakeFiles/KinesisVideoProducer.dir/build.make:62: recipe for target 'dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/CMakeFiles/KinesisVideoProducer.dir/src/Auth.cpp.o' failed
make[2]: *** [dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/CMakeFiles/KinesisVideoProducer.dir/src/Auth.cpp.o] Error 1
In file included from /path/to/my/repo/dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/src/DefaultCallbackProvider.h:11:0,
                 from /path/to/my/repo/dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/src/DefaultCallbackProvider.cpp:3:
/path/to/my/repo/dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/src/Auth.h:12:10: fatal error: Logger.h: No such file or directory
 #include <Logger.h>
          ^~~~~~~~~~
compilation terminated.
In file included from /path/to/my/repo/dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/src/DefaultCallbackProvider.h:11:0,
                 from /path/to/my/repo/dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/src/CachingEndpointOnlyCallbackProvider.h:10,
                 from /path/to/my/repo/dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/src/CachingEndpointOnlyCallbackProvider.cpp:3:
/path/to/my/repo/dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/src/Auth.h:12:10: fatal error: Logger.h: No such file or directory
 #include <Logger.h>
          ^~~~~~~~~~
compilation terminated.
dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/CMakeFiles/KinesisVideoProducer.dir/build.make:134: recipe for target 'dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/CMakeFiles/KinesisVideoProducer.dir/src/DefaultCallbackProvider.cpp.o' failed
make[2]: *** [dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/CMakeFiles/KinesisVideoProducer.dir/src/DefaultCallbackProvider.cpp.o] Error 1
dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/CMakeFiles/KinesisVideoProducer.dir/build.make:86: recipe for target 'dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/CMakeFiles/KinesisVideoProducer.dir/src/CachingEndpointOnlyCallbackProvider.cpp.o' failed
make[2]: *** [dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/CMakeFiles/KinesisVideoProducer.dir/src/CachingEndpointOnlyCallbackProvider.cpp.o] Error 1
In file included from /path/to/my/repo/dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/src/DefaultCallbackProvider.h:11:0,
                 from /path/to/my/repo/dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/src/KinesisVideoProducer.h:9,
                 from /path/to/my/repo/dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/src/KinesisVideoProducer.cpp:1:
/path/to/my/repo/dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/src/Auth.h:12:10: fatal error: Logger.h: No such file or directory
 #include <Logger.h>
          ^~~~~~~~~~
compilation terminated.
dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/CMakeFiles/KinesisVideoProducer.dir/build.make:206: recipe for target 'dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/CMakeFiles/KinesisVideoProducer.dir/src/KinesisVideoProducer.cpp.o' failed
make[2]: *** [dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/CMakeFiles/KinesisVideoProducer.dir/src/KinesisVideoProducer.cpp.o] Error 1
In file included from /path/to/my/repo/dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/src/DefaultCallbackProvider.h:11:0,
                 from /path/to/my/repo/dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/src/KinesisVideoProducer.h:9,
                 from /path/to/my/repo/dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/src/KinesisVideoStream.h:10,
                 from /path/to/my/repo/dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/src/KinesisVideoStream.cpp:2:
/path/to/my/repo/dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/src/Auth.h:12:10: fatal error: Logger.h: No such file or directory
 #include <Logger.h>
          ^~~~~~~~~~
compilation terminated.
dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/CMakeFiles/KinesisVideoProducer.dir/build.make:230: recipe for target 'dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/CMakeFiles/KinesisVideoProducer.dir/src/KinesisVideoStream.cpp.o' failed
make[2]: *** [dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/CMakeFiles/KinesisVideoProducer.dir/src/KinesisVideoStream.cpp.o] Error 1
CMakeFiles/Makefile2:90: recipe for target 'dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/CMakeFiles/KinesisVideoProducer.dir/all' failed
make[1]: *** [dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/CMakeFiles/KinesisVideoProducer.dir/all] Error 2
Makefile:129: recipe for target 'all' failed
make: *** [all] Error 2
```

When I print the elements of the `INCLUDE_DIRECTORIES` variable of the `amazon-kinesis-video-streams-producer-sdk-cpp` directory, here is the result (pay special attention to the final few lines):

```
-- kvs includes='/path/to/my/repo/src/includes'
-- kvs includes='/usr/include'
-- kvs includes='/usr/include/boost'
-- kvs includes='/opt/nvidia/deepstream/deepstream-4.0/sources/includes'
-- kvs includes='/usr/local/cuda-10.0/include'
-- kvs includes='/usr/include/glib-2.0'
-- kvs includes='/usr/lib/aarch64-linux-gnu/glib-2.0/include'
-- kvs includes='/usr/include/glib-2.0'
-- kvs includes='/usr/lib/aarch64-linux-gnu/glib-2.0/include'
-- kvs includes='/usr/include/json-glib-1.0'
-- kvs includes='/usr/include/glib-2.0'
-- kvs includes='/usr/lib/aarch64-linux-gnu/glib-2.0/include'
-- kvs includes='/usr/include/uuid'
-- kvs includes='/usr/local/include'
-- kvs includes='/usr/include/gstreamer-1.0'
-- kvs includes='/usr/include/glib-2.0'
-- kvs includes='/usr/lib/aarch64-linux-gnu/glib-2.0/include'
-- kvs includes='/usr/include/gstreamer-1.0'
-- kvs includes='/usr/include/glib-2.0'
-- kvs includes='/usr/lib/aarch64-linux-gnu/glib-2.0/include'
-- kvs includes='/usr/include/gstreamer-1.0'
-- kvs includes='/usr/include/orc-0.4'
-- kvs includes='/usr/include/gstreamer-1.0'
-- kvs includes='/usr/include/glib-2.0'
-- kvs includes='/usr/lib/aarch64-linux-gnu/glib-2.0/include'
-- kvs includes='/usr/include/opencv4'
-- kvs includes='/path/to/my/repo/dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/open-source/local/include'
-- kvs includes='/path/to/my/repo/dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/dependency/libkvscproducer/kvscproducer-src/dependency/libkvspic/kvspic-src/src/client/include'
-- kvs includes='/path/to/my/repo/dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/dependency/libkvscproducer/kvscproducer-src/dependency/libkvspic/kvspic-src/src/common/include'
-- kvs includes='/path/to/my/repo/dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/dependency/libkvscproducer/kvscproducer-src/dependency/libkvspic/kvspic-src/src/heap/include'
-- kvs includes='/path/to/my/repo/dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/dependency/libkvscproducer/kvscproducer-src/dependency/libkvspic/kvspic-src/src/mkvgen/include'
-- kvs includes='/path/to/my/repo/dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/dependency/libkvscproducer/kvscproducer-src/dependency/libkvspic/kvspic-src/src/state/include'
-- kvs includes='/path/to/my/repo/dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/dependency/libkvscproducer/kvscproducer-src/dependency/libkvspic/kvspic-src/src/trace/include'
-- kvs includes='/path/to/my/repo/dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/dependency/libkvscproducer/kvscproducer-src/dependency/libkvspic/kvspic-src/src/utils/include'
-- kvs includes='/path/to/my/repo/dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/dependency/libkvscproducer/kvscproducer-src/dependency/libkvspic/kvspic-src/src/view/include'
-- kvs includes='/path/to/my/repo/dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/dependency/libkvscproducer/kvscproducer-src/src/include'
-- kvs includes='/path/to/my/repo/dependencies/amazon-kinesis-video-streams-producer-sdk-cpp/open-source/include'
-- kvs includes='/path/to/my/repo/src'
-- kvs includes='/path/to/my/repo/src/credential-providers'
-- kvs includes='/path/to/my/repo/src/common'
-- kvs includes='/path/to/my/repo/src/JNI/include'
```

The last few lines seem to indicate that `CMAKE_SOURCE_DIR` is used instead of `CMAKE_CURRENT_SOURCE_DIR`, which means that this project won't be successfully built when included as a subdirectory. I'll open up a PR to add a fix.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -39,7 +39,7 @@ else()
   set(LINKAGE SHARED)
 endif()
 
-set(KINESIS_VIDEO_PRODUCER_CPP_SRC "${CMAKE_SOURCE_DIR}")
+set(KINESIS_VIDEO_PRODUCER_CPP_SRC "${CMAKE_CURRENT_SOURCE_DIR}")
 set(KINESIS_VIDEO_OPEN_SOURCE_SRC ${CMAKE_CURRENT_SOURCE_DIR}/open-source)
 
 message(STATUS "Kinesis Video Cpp Producer path is ${KINESIS_VIDEO_PRODUCER_CPP_SRC}")
@@ -50,7 +50,7 @@ if(NOT EXISTS ${KINESIS_VIDEO_PRODUCER_CPP_SRC})
 endif()
 
 # pass ca cert location to sdk
-add_definitions(-DKVS_CA_CERT_PATH="${CMAKE_SOURCE_DIR}/certs/cert.pem")
+add_definitions(-DKVS_CA_CERT_PATH="${CMAKE_CURRENT_SOURCE_DIR}/certs/cert.pem")
 add_definitions(-DCMAKE_DETECTED_CACERT_PATH)
 
 if(BUILD_DEPENDENCIES)

